core/src/main/kotlin/com/gitlab/kordlib/core/entity/channel/GuildChannel.kt

```diff
@@ -52,28 +52,25 @@ interface GuildChannel : Channel, GuildChannelBehavior {
 
         val base = member.getPermissions()
 
-        if (Permission.Administrator in base) return Permissions { +Permission.All }
-
+        if (Permission.Administrator in base) return Permissions(Permission.All)
+        var result = base
         val everyoneOverwrite = getPermissionOverwritesForRole(guildId)
         val roleOverwrites = member.roleIds.mapNotNull { getPermissionOverwritesForRole(it) }
         val memberOverwrite = getPermissionOverwritesForMember(memberId)
 
-        return Permissions {
-            +base
-            everyoneOverwrite?.let {
-                +it.allowed
-                -it.denied
-            }
-            roleOverwrites.map {
-                +it.allowed
-                -it.denied
-            }
-            memberOverwrite?.let {
-                +it.allowed
-                -it.denied
-            }
+        if (everyoneOverwrite != null) {
+            result += everyoneOverwrite.allowed
+            result -= everyoneOverwrite.denied
         }
-
+        roleOverwrites.forEach {
+            result += it.allowed
+            result -= it.denied
+        }
+        if (memberOverwrite != null) {
+            result += memberOverwrite.allowed
+            result -= memberOverwrite.denied
+        }
+        return result
```

Comment:
This also seems like a good argument for not deleting the DSL, I find this less readable and less performant.

Reply:
This is was the idea behind making `code` property mutable. I also considered making it internal as well. I guess I still need to work around the bikeshidding habit of mine.